### PR TITLE
osx_say: add espeak support, rename module to say

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
- add support for `espeak` (and `say` GNUstep tool)
- rename module
- handle `check_mode`

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
say

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 0324d5e850) last updated 2017/11/20 02:15:48 (GMT +200)
```

### CODE REVIEW

File: lib/ansible/modules/notification/say.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        supports_check_mode=True
+    )
+
+    msg = module.params['msg']
+    voice = module.params['voice']
+
+    executable = module.get_bin_path('say')
+    if not executable:
+        executable = module.get_bin_path('espeak')
+    elif get_platform() != 'Darwnin':
```
Comment (resmo) [inline lib/ansible/modules/notification/say.py]: typo

Comment (pilou-) [reply lib/ansible/modules/notification/say.py]: Fixed, thanks!

**Comment by mscherer** [comment]
shipit

File: lib/ansible/modules/notification/say.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: say
+version_added: "1.2"
+short_description: Makes a computer to speak.
+description:
+   - makes acomputer speak! Amuse your friends, annoy your coworkers!
```
Comment (jhawkesworth) [inline lib/ansible/modules/notification/say.py]: typo: acomputer -> a computer

**Comment by jhawkesworth** [comment]
Just a tiny typo but other than that it looks good to me.

**Comment by pilou-** [comment]
@jhawkesworth updated, thanks !

**Comment by mscherer** [comment]
shipit

**Comment by pilou2** [comment]
bot_status

**Comment by pilou-** [comment]
bot_status

**Comment by pilou-** [comment]
@mscherer: PR was rebased in order to fix a conflict :)

@resmo could you please review this pull-request ?

**Comment by resmo** [comment]
LGTM based on code review

shipit

**Comment by gundalow** [comment]
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

**Comment by mscherer** [comment]
shipit

**Comment by pilou-** [comment]
@resmo sorry to bother you again, could you please review this one again ?

**Comment by resmo** [comment]
shipit

**Comment by gundalow** [comment]
Could you please update github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guide_2.5.rst as well

**Comment by pilou-** [comment]
@gundalow updated

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thanks, merged into `devel`

**Pull request merged**